lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -642,4 +572,129 @@ defmodule Trento.Domain.SapSystem do
         false
     end)
   end
+
+  defp abap_or_messageserver?(features) do
+    String.contains?(features, ["MESSAGESERVER", "ABAP"])
+  end
+
+  defp abap_and_messageserver?(features) do
+    String.contains?(features, "ABAP") and String.contains?(features, "MESSAGESERVER")
+  end
+
+  defp maybe_emit_sap_system_registered_event(
+         %SapSystem{sid: nil, application: %Application{instances: instances}},
+         %RegisterApplicationInstance{
+           sap_system_id: sap_system_id,
+           sid: sid,
+           tenant: tenant,
+           db_host: db_host,
+           features: features,
+           health: health
+         }
+       ) do
+    instances_features =
+      instances
+      |> Enum.map(& &1.features)
```

Comment:
Hey @CDimonaco , I think if you use the suggestion by @arbulu89 in https://github.com/trento-project/web/pull/1358/files#r1178679263 you wouldn't need to do this in the first place